internal/sddc/dns_subresource.go

```diff
@@ -0,0 +1,66 @@
+/*
+ *  Copyright 2023 VMware, Inc.
+ *    SPDX-License-Identifier: MPL-2.0
+ */
+
+package sddc
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	utils "github.com/vmware/terraform-provider-vcf/internal/resource_utils"
+	"github.com/vmware/vcf-sdk-go/models"
+)
+
+func GetDnsSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		Required: true,
+		MaxItems: 1,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"domain": {
+					Type:        schema.TypeString,
+					Description: "Tenant domain. Example: rainpole.io",
+					Required:    true,
+				},
+				"name_server": {
+					Type:         schema.TypeString,
+					Description:  "Primary nameserver to be configured for vCenter/PSC/ESXi's/NSX. Example: 172.0.0.4",
+					Optional:     true,
+					ValidateFunc: validation.IsIPAddress,
+				},
+				"secondary_name_server": {
+					Type:         schema.TypeString,
+					Description:  "Secondary nameserver to be configured for vCenter/PSC/ESXi's/NSX. Example: 172.0.0.5",
+					Optional:     true,
+					ValidateFunc: validation.IsIPAddress,
+				},
+				"subdomain": {
```

Comment:
subdomain is very legacy - v2.x.

`dns_namespace` would be more accurate.

Reply:
Not even sure why this is still here though. Esp since `domain` would be eq `dns_namespace`..

Reply:
actually, it's the opposite (for some historical reasons).. subdomain takes precedence if both of the fields are provided as input. So here we can decide to just expose one field for DNS domain and map it to the subdomain field in the CB REST API.

Reply:
I'm leaving only the "domain" attribute, but I'll set it's value to the Subdomain property in the DNSSpec.